app/src/main/java/org/mozilla/focus/widget/Cursor.java

```diff
@@ -80,19 +85,30 @@ public void moveCursor(Direction direction) {
 
     public void stopMoving(Direction direction) {
         activeDirections.remove(direction);
+
         if (activeDirections.size() == 0) {
             handler.removeCallbacks(tick);
             moving = false;
             speed = 0;
+
+            animate().alpha(VIEW_MIN_ALPHA)
+                    .setDuration(CURSOR_ANIMATION_DURATION)
+                    .setStartDelay(CURSOR_HIDE_AFTER_TIME);
         }
     }
 
     private void move() {
+        if (getAlpha() != VIEW_MAX_ALPHA) {
```

Comment:
I feel like this should be in `moveCursor`, in the `!moving` branch, so we're not calling this every frame:
```java
if (!moving()) {
    isMoving = true;
    setAlpha(VIEW_MAX_ALPHA);
    animate().cancel();
}
```

- Is there a reason this is done in `move` (i.e. each frame)?
- Why do we check the alpha? It feels like the alpha comparison here is being used as a proxy for `if (!moving)` (in my code snippet)
- Why are `setAlpha` and `cancel()` are being called from different branches?

Reply:
> * Why are setAlpha and cancel() are being called from different branches?

I added a couple of comments to the code to clarify. I think it could probably be more clear with the code, but I was struggling on coming up with a better solution